src/vtrasterizer/DecorationRenderer.cpp

```diff
@@ -139,16 +139,15 @@ auto DecorationRenderer::createTileData(Decorator decoration, atlas::TileLocatio
             // TODO (default to Underline for now)
             [[fallthrough]];
         case Decorator::Underline: {
-            auto const thicknessHalf = max(1u, unsigned(ceil(underlineThickness() / 2.0)));
-            auto const thickness = thicknessHalf * 2;
-            auto const y0 = max(0, (underlinePosition() - static_cast<int>(thicknessHalf)));
+            auto const thickness = max(1u, unsigned(ceil(underlineThickness() / 2.0)));
+            auto const y0 = max(0, (underlinePosition() - static_cast<int>(thickness)));
             auto const height = vtbackend::Height(y0 + thickness);
             auto const imageSize = ImageSize { width, height };
             return create(imageSize, [&]() -> atlas::Buffer {
                 auto image = atlas::Buffer(imageSize.area(), 0);
                 for (unsigned y = 1; y <= thickness; ++y)
-                    for (unsigned x = 0; x < unbox(width); ++x)
-                        image[(*height - y0 - y) * *width + x] = 0xFF;
+                    for (auto x: crispy::times(unbox(width)))
```

Comment:
actually, I think we can use `std::views::iota()` by now. what do you think? (also, `for (auto const x: ...` rather than `for (auto x: ...`.

Analog the equivalent parts below. :)